src/app/dashboard/components/filters/form.tsx

```diff
@@ -1,47 +1,108 @@
 import { zodResolver } from '@hookform/resolvers/zod/dist/zod';
-import { Button } from '@mui/material';
-import React from 'react';
+import { Dayjs } from 'dayjs';
 import { useForm, useWatch } from 'react-hook-form';
 import styled from 'styled-components';
+import { NumberParam, StringParam, useQueryParams } from 'use-query-params';
 
 import Form from 'components/form';
 import DateInput from 'components/form/dateInput';
+import NumberInput from 'components/form/numberInput';
 
 import { FiltersForm, FiltersFormSchema } from './schema';
-import { getFilterFormDefaultValues, minDateBoundary, today } from './utils';
+import { minDateBoundary, today } from './utils';
+import { useDebounce } from 'use-debounce';
+import { useEffect } from 'react';
+
+interface Props {
+  setStartDateHandle: (childData: Dayjs) => void;
+  setEndDateHandle: (childData: Dayjs) => void;
+  setAggregateHandle: (aggregate: number) => void;
+  startDate: Dayjs;
+  endDate: Dayjs;
+  aggregate: number;
+}
+
+const StockChartFiltersForm = (props: Props) => {
+  const [query, setQuery] = useQueryParams({
+    date_from: StringParam,
+    date_to: StringParam,
+    aggregate: NumberParam,
+  });
+  query;
 
-const StockChartFiltersForm: React.FC = () => {
   const methods = useForm<FiltersForm>({
-    defaultValues: getFilterFormDefaultValues(),
+    defaultValues: {
+      startDate: props.startDate,
+      endDate: props.endDate,
+      aggregate: props.aggregate,
+    },
```

Comment:
No i teraz zaczynamy zabawę z aktualizowaniem wartości formularza oraz z aktualizowaniem query.

1. Domyślne wartości są pobierane z `query`

```suggestion
    defaultValues: {
      startDate: dayjs(query.date_from),
      endDate: dayjs(query.date_to),
      aggregate: query.aggregate,
    },
```